Improve errors messages thrown from a `transform` closure (#115)
apple/swift-argument-parser: Documentation/02 Arguments, Options, and Flags.md

}
```

> Throw an error from the `transform` function to indicate that the user provided an invalid value for that type. (See: [Handling Transform Errors](./05%20Validation%20and%20Errors.md#handling-transform-errors) in [Validation and Errors](./05%20Validation%20and%20Errors.md) for details about customising `transform` function errors.)

Reviewer: Nits: This sentence doesn't need to be parenthesized, and let's use the American spelling here.
```suggestion
Throw an error from the `transform` function to indicate that the user provided an invalid value for that type. See [Handling Transform Errors](./05%20Validation%20and%20Errors.md#handling-transform-errors) for more about customizing `transform` function errors.
```
